sdks/python/apache_beam/io/datastore/v1/datastoreio.py

```diff
@@ -285,3 +286,105 @@ def get_estimated_num_splits(project, namespace, query, datastore):
       num_splits = ReadFromDatastore._NUM_QUERY_SPLITS_MIN
 
     return max(num_splits, ReadFromDatastore._NUM_QUERY_SPLITS_MIN)
+
+
+class Mutate(PTransform):
+  """A ``PTransform`` that writes mutations to Cloud Datastore.
+
+  Only idempotent Datastore mutation operations (upsert and delete) are
```

Comment:
We don't need to because this is not supposed to be used directly, Write and Delete transforms enforces. (Renamed it _Mutate to be explicit)